lib/connectors_sdk/share_point/adapter.rb

```diff
@@ -29,18 +29,36 @@ class FileGraphItem < Office365::Adapter::FileGraphItem
         def self.convert_id_to_es_id(id)
           ConnectorsSdk::SharePoint::Adapter.share_point_id_to_es_id(id)
         end
+
+        private
+
+        def get_path(item)
+          item.site_name.present? ? "/sites/#{item.site_name}#{super}" : super
+        end
```

Comment:
I am confused why we have to write three times the same method here. I feel like we could have that in a single place

Reply:
I agree with all of the above.

Reply:
Well, it still looks like a lot of duplication. This kind of duplication is usually a smell of a design issue. 

Why can't you use `include` btw? (multiple inheritance)

All those classes could include a `SitePrefix` class that implements `get_path` so you remove that duplication and don't have to copy over the same method everywhere.

Reply:
Introduced module `SitePrefix`